internal/distro/distro.go

```diff
@@ -108,6 +108,9 @@ type ImageType interface {
 	// Returns the names of the pipelines that create the image.
 	PayloadPipelines() []string
 
+	// Returns the package set names safe to install custom packages via custom repositories.
+	PackageSetKeys() []string
```

Comment:
The name `PackageSetKeys` is a bit too generic imo, but coming up with a good name here is hard.  Since we call it Build/Payload, maybe we can call this `PayloadPackageSets() []string` @achilleas-k wdyt?